Index: pylib/gyp/generator/make.py
===================================================================
--- pylib/gyp/generator/make.py	(revision 1884)
+++ pylib/gyp/generator/make.py	(working copy)
@@ -316,8 +316,9 @@
 # careful here to use the flags that ccache and distcc can understand.
 # We write to a dep file on the side first and then rename at the end
 # so we can't end up with a broken dep file.
-depfile = $(depsdir)/$(call replace_spaces,$@).d
-DEPFLAGS = -MMD -MF $(depfile).raw
+depname = $(depsdir)/$(call replace_spaces,$@)
+depfile = $(depname).d
+DEPFLAGS = -Wp,-MMD,$(depfile).raw,-MT,$(depname)
 
 # We have to fixup the deps output in a few ways.
 # (1) the file output should mention the proper .o file.
